source/plugins/ruby/out_mdm.rb

```diff
@@ -106,6 +108,9 @@ def start
             @cluster_identity = ArcK8sClusterIdentity.new
             @cached_access_token = @cluster_identity.get_cluster_identity_token
           else
+            # azure json file only used for aks and doesnt exist in non-azure envs
```

Comment:
what is the actual issue due to which this fix is needed ? When i check the cidev cluster i see custom metrics . Can u explain the problem for this fix ?

Reply:
> what is the actual issue due to which this fix is needed ? When i check the cidev cluster i see custom metrics . Can u explain the problem for this fix ?

>  as you can see, code comment has the explanation  " # azure json file only used for aks and doesnt exist in non-azure envs". since azure.json file doesnt exist on-prem k8s cluster so we were crashing.  we only use azure json file for aks so moved under code path.

Reply:
so we added metrics for non-aks and tested in aks ?